app/users/get-user/getUserInteractor.js

```diff
@@ -0,0 +1,30 @@
+const NotFound = require('../../errors/notFound');
+const ValidationError = require('../../errors/validationError');
+
+class GetUserInteractor {
+    constructor ({presenter, validator, userRepository}) {
+        this.presenter = presenter;
+        this.validator = validator;
+        this.userRepository = userRepository;
+    }
+
+    async execute({id}) {
+        const errors = this.validator.validate({id});
+
+        if (errors.length > 0) {
+            this.presenter.presentFailure( new ValidationError(errors) );
+            return;
+        }
+
+        const user = await this.userRepository.findByID({id});
+
+        if (!user) {
+            this.presenter.presentFailure( new NotFound(`User with ${id} was not found`));
+            return;
+        }
+
+        this.presenter.presentSuccess(user);
```

Comment:
It is nice, but from here we should start to implement such pattern:
anytime we need to respond with smth, we should actually prepare it befor sending.
Why we should: data we got from DB is pretty raw, and may content some app logic things, or smth we dont want client to know about. In this case we should implement `ResponseBuilder` for each case, that need to respond with data. This builder should prepare entity (or anything else) before it would be sent. Yeah, basically, it's `build()` method whould be just do smth like `return {id: entity.getId(), name: entity.getName()..}` except the fields we dont want to display (its OK when every field should be displayed too)